ext/mysql2/result.c

```diff
@@ -826,11 +861,19 @@ static VALUE rb_mysql_result_each(int argc, VALUE * argv, VALUE self) {
   }
 
   symbolizeKeys = RTEST(rb_hash_aref(opts, sym_symbolize_keys));
-  asArray       = rb_hash_aref(opts, sym_as) == sym_array;
+  rowsAs        = AS_HASH;
```

Comment:
Nit: make this the `else` case in the block below. (or the `default` case, and convert to a `switch` statement)

Reply:
Done.